packages/flutter/lib/src/material/colors.dart

```diff
@@ -2,7 +2,76 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:ui' show Color;
+import 'dart:ui' show Color, hashValues;
+
+class _ColorSwatch extends Color {
+  const _ColorSwatch(int primary, this.swatch) : super(primary);
+
+  /// The shades of this color defined by material design.
+  final Map<int, Color> swatch;
+
+  /// Returns an element of the [swatch] table.
+  Color operator [](int index) => swatch[index];
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! _ColorSwatch)
```

Comment:
compare runtimeType rather than using `is`, otherwise it's possible a subclass will be `==` in one direction but not in teh other (yes we get this wrong all over the place)